extensions/amp-ad-exit/0.1/filters/click-delay.js

```diff
@@ -21,32 +21,34 @@ export class ClickDelayFilter extends Filter {
   /**
    * @param {string} name The user-defined name of the filter.
    * @param {!../config.ClickDelayConfig} spec
+   * @param {!Window} win
    */
-  constructor(name, spec) {
+  constructor(name, spec, win) {
     super(name);
-    user().assert(isValidClickDelaySpec(spec), 'Invalid ClickDelay spec');
+    user().assert(spec.type == FilterType.CLICK_DELAY &&
+      typeof spec.delay == 'number' && spec.delay > 0,
+    'Invalid ClickDelay spec');
+    user().assert(!spec.startTimingEvent ||
+      (win['performance'] && win['performance']['timing']),
+    'Browser does not support performance timing');
```

Comment:
Should this be a fatal problem? Should it be the responsibility of the creative author to anticipate which browsers their creative will serve to and know which web platform features are available? I'd rather treat this property as a best-effort feature upgrade.

Reply:
If performance timing isn't available, we should probably fall back to 1s from when we first got to run our code, no?

Reply:
Done - falls back to using time interval when extension executed